src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -892,5 +954,58 @@ public int hashCode() {
     public String toString() {
         return "transaction " + hash.toString();
     }
-}
 
+    /**
+     * Puts the TVM in cache
+     * @param tangle Tangle
+     * @param transactionViewModel The tvm to cache
+     * @param hash the hash of the tvm
+     * @throws Exception Exception
+     */
+    private static void cachePut(Tangle tangle, TransactionViewModel transactionViewModel, Hash hash) throws Exception {
+        Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
+        if (cache.getSize() == cache.getConfiguration().getMaxSize()) {
+            cacheEvict(tangle);
```

Comment:
I would expect the cache itself would know to do that

Reply:
1. Don't want to pass `tangle` to the cache.
2. Eviction is tightly coupled to `TransactionViewModels`.  We need the cache to be oblivious to these ViewModels.
Only option, for now, is to do it here.

Reply:
I keep on seeing the ugly design, and I want to make it better but is impossible :-(
I guess it would be possible if we do the ViewModel interface maybe, but we won't do it now ofcourse